### PR TITLE
Workaround caching problems with java-gradle-plugin

### PR DESCRIPTION
**Pull request by wolfs**
When the java-gradle-plugin is applied, it has too bad side effects for caching:

for tests in buildSrc, the pluginUnderTestMetadata file is added to the test runtime classpath here. This file contains absolute paths.
The properties file written out by GeneratePluginDescriptors contains system dependent line separators, which breaks caching of any custom task between windows and linux.

This PR works around both of these problems until they are fixed in Gradle core.

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -1,3 +1,11 @@
+
+import org.gradle.api.JavaVersion
```
Comment (eskatos) [inline buildSrc/build.gradle.kts]: :x: Most of these imports are implicit and not needed

Comment (eskatos) [reply buildSrc/build.gradle.kts]: And the imports should be moved below the license header

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -84,3 +92,26 @@ if (!isCiServer || System.getProperty("enableCodeQuality")?.toLowerCase() == "tr
 }
 
 apply { from("../gradle/ciReporting.gradle") }
+
+// Workaround caching problems with 'java-gradle-plugin'
+// vvvvv
```
Comment (eskatos) [inline buildSrc/build.gradle.kts]: :x: This line should probably be removed

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (wolfs) [reply buildSrc/build.gradle.kts]: How do I tell that to IDEA?

Comment (wolfs) [reply buildSrc/build.gradle.kts]: IOW: I had "organize imports" switched on and that is what I ended up with.

Comment (eskatos) [reply buildSrc/build.gradle.kts]: If using project import then the implicit imports should be handled by IntelliJ.

I don't use the "on the fly imports" feature and, using e.g. `JavaVersion` doesn't add any import statement as it is already available via implicit imports. When triggering "organize imports" it doesn't add it either.

image

What are your settings?

About putting the imports below the licence header, I don't know how to tell that to IntelliJ.

Comment (wolfs) [reply buildSrc/build.gradle.kts]: "Optimize imports on the fly" seems to be the culprit.

Comment (wolfs) [reply buildSrc/build.gradle.kts]: I fixed the import statements manually, please have another look.

I think Jetbrains should fix the problems with "Optimize imports on the fly" and Kotlin build scripts. I had a hard time importing the classes anyway, the popup didn't show the corresponding classes (Alt-Return). There may also be some problems there.

Comment (eskatos) [reply buildSrc/build.gradle.kts]: @wolfs I just opened https://youtrack.jetbrains.com/issue/KT-21981, please upvote!

**Pull request merged**